src/plcc/parse_spec/parse_syntactic_spec/parse_syntactic_spec_test.py

```diff
@@ -0,0 +1,353 @@
+from pytest import raises, mark, fixture
+from .parse_syntactic_spec import parse_syntactic_spec
+from .parse_syntactic_structs import (
+    CapturingTerminal,
+    RepeatingSyntacticRule,
+    StandardSyntacticRule,
+    LhsNonTerminal,
+    RhsNonTerminal,
+    Terminal,
+    MalformedBNFError,
+)
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+
+
+def test_None_yields_nothing():
+    assert list(parse_syntactic_spec(None)) == []
+
+
+def test_empty_yields_nothing():
+    assert list(parse_syntactic_spec([])) == []
+
+
+def test_divider_yields_nothing():
+    lines = [makeDivider()]
+    assert list(parse_syntactic_spec(lines)) == []
+
+
+def test_blank_rule():
+    lines = [makeDivider(), makeLine("<noun> ::=")]
+    expected = [
+        StandardSyntacticRule(makeLine("<noun> ::="), LhsNonTerminal("noun"), [])
+    ]
+    assert list(parse_syntactic_spec(lines)) == expected
```

Comment:
Instances where you use the same line multiple times within the same test can be assigned to a variable, so that if anything needs to be changed, you will only need to change it in one place rather than everywhere, this can be applied to multiple tests below:
```
line = makeLine("<noun> ::=")
```